pylib/Tools/Build/Autotools.py

```diff
@@ -197,6 +199,16 @@ def execute(self, log, keyvals, testDef):
                     # this is a multistep operation, and so we need to
                     # retain the output from each step in the log
                     log['autogen'] = (stdout, stderr)
+
+                    # Use mpi4py to get MPI information for IUDatabase plugin
+                    log['mpi_name'] = MPI.get_vendor()[0]
```

Comment:
mpi4py get_vendor() function simply calls a C function using something like cython. The C function then makes use of macros defined at compile time that tells it which MPI version is running.

My best solution using pure python would be to parse the output of "mpirun --version", which isn't as clean of a solution as what mpi4py does since "mpirun --version" doesn't have a standard output.